Highlighting Error: not a valid line for pointer

when running sonar-perl against this file (text replaced):
```
#!/usr/bin/perl

use strict;

sub usage {
    print <<EOF;
Usage: $0

Lorem ipsum dolor sit amet, consectetur adipiscing elit,
sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit
in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
Excepteur sint occaecat cupidatat non proident,
sunt in culpa qui officia deserunt mollit anim id est laborum

EOF

    exit;
}
```

I get this error:
java.lang.IllegalArgumentException: Unable to highlight file $file
Caused by: java.lang.IllegalArgumentException: 27 is not a valid line for pointer. File $file has 21 line(s)

I haven't exactly been able to figure out what is causing it, but I do know it depends on the amount of lines. So for example, this:

```
#!/usr/bin/perl

use strict;

sub usage {
    print <<EOF;
Usage: $0

Lorem ipsum dolor sit amet, consectetur adipiscing elit,

EOF
    exit;
}
```

doesn't work,

but this 

```
#!/usr/bin/perl

use strict;

sub usage {
    print <<EOF;
Usage: $0

Lorem ipsum dolor sit amet, consectetur adipiscing elit,

EOF

    exit;
}
```

does work.

I have managed to find examples where a couple of lines worked, and some where they did not. I am not completely sure what is going on. All I have been able to figure out is that the amount of lines matters, and that the specific text does not.

--- a/its/plugin/projects/metrics/lib/heredoc.pl
+++ b/its/plugin/projects/metrics/lib/heredoc.pl
@@ -1,0 +1,20 @@
+#!/usr/bin/perl
+
+use strict;
+
+sub usage {
+    print <<EOF;
+Usage: $0
+
+Lorem ipsum dolor sit amet, consectetur adipiscing elit,
+sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
+nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit
+in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
+Excepteur sint occaecat cupidatat non proident,
+sunt in culpa qui officia deserunt mollit anim id est laborum
+
+EOF
+
+    exit;
+}

--- a/its/plugin/src/test/java/com/github/sonarperl/it/ProjectMetricsIntegrationTest.java
+++ b/its/plugin/src/test/java/com/github/sonarperl/it/ProjectMetricsIntegrationTest.java
@@ -56,13 +56,13 @@ public class ProjectMetricsIntegrationTest {
     @Test
     public void project_level() {
       // Size
-      assertThat(wsClient.getProjectMeasure("ncloc")).isEqualTo(15);
-      assertThat(wsClient.getProjectMeasure("lines")).isEqualTo(59);
-      assertThat(wsClient.getProjectMeasure("files")).isEqualTo(1);
-      assertThat(wsClient.getProjectMeasure("functions")).isEqualTo(3);
+      assertThat(wsClient.getProjectMeasure("ncloc")).isEqualTo(29);
+      assertThat(wsClient.getProjectMeasure("lines")).isEqualTo(79);
+      assertThat(wsClient.getProjectMeasure("files")).isEqualTo(2);
+      assertThat(wsClient.getProjectMeasure("functions")).isEqualTo(4);
       assertThat(wsClient.getProjectMeasure("classes")).isEqualTo(1);
       // Documentation
-      assertThat(wsClient.getProjectMeasure("comment_lines")).isEqualTo(26);
+      assertThat(wsClient.getProjectMeasure("comment_lines")).isEqualTo(27);
     }
 
     @Test
@@ -87,6 +87,8 @@ public class ProjectMetricsIntegrationTest {
       assertThat(wsClient.getFileMeasure("lib/Sample/Project.pm", "classes")).isEqualTo(1);
       // Documentation
       assertThat(wsClient.getFileMeasure("lib/Sample/Project.pm", "comment_lines")).isEqualTo(26);
+      // heredoc
+      assertThat(wsClient.getFileMeasure("lib/heredoc.pl", "lines")).isEqualTo(20);
     }
 
 }

--- a/perl-frontend/src/main/java/com/github/sonarperl/lexer/HeredocChannel.java
+++ b/perl-frontend/src/main/java/com/github/sonarperl/lexer/HeredocChannel.java
@@ -53,6 +53,8 @@ public class HeredocChannel extends Channel<Lexer> {
     }
 
     private boolean consumeUntil(CodeReader code, Lexer output) {
+        int line = code.getLinePosition();
+        int column = code.getColumnPosition();
 
         StringBuilder sb = new StringBuilder();
         Matcher matcher = Pattern.compile(".*?(?=\\Q" + lookFor + "\\E)", Pattern.DOTALL).matcher("");
@@ -66,8 +68,8 @@ public class HeredocChannel extends Channel<Lexer> {
           .setType(PerlTokenType.STRING)
           .setValueAndOriginalValue(sb.toString())
           .setURI(output.getURI())
-          .setLine(code.getLinePosition())
-          .setColumn(code.getColumnPosition())
+          .setLine(line)
+          .setColumn(column)
           .build();
 
         output.addToken(token);

